File: tutorials/testing/c++/tut_worldmap_p4.cpp
#include <string>
#include <vector>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "WorldMap.h"

#include "SymbolCollection.h"
#include "Circle.h"
#include "Polyline.h"
#include "Rectangle.h"
#include "Text.h"

using namespace bridges;

int main(int argc, char **argv) {

	// create bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

	// set title
	bridges.setTitle("Drawing a World Map");

	// set description
	bridges.setDescription("This example illustrates overlaying symbols to label cities on a world map overlay");

	// create a data source
	DataSource ds(&bridges);


	// add symbols 
	SymbolCollection sc;

	// plot a bunch of cities
	Circle c(-80.8431, 35.2271, 1.8);
		c.setStrokeColor("red");
		c.setFillColor("cyan");
	sc.addSymbol(c);

	Text l;
		float loc[2] = {-79.8431, 35.2271};
		l.setAnchorLocation(loc);
		l.setFontSize(10.0f);
		l.setStrokeWidth(1.0f);
		l.setText("Charlotte");
		l.setStrokeColor("purple");
		l.setOpacity(1.0f);
    sc.addSymbol(l);

	Circle c2(2.3514, 48.8576, 1.);
		c2.setStrokeColor("red");
		c2.setFillColor("cyan");
	sc.addSymbol(c2);

	Text l2;
		float loc2[2] = {3.3514, 48.8576};
		l2.setAnchorLocation(loc2);
		l2.setFontSize(10.0f);
		l2.setStrokeWidth(1.0f);
		l2.setText("Paris");
		l2.setStrokeColor("purple");
		l2.setOpacity(1.0f);
    sc.addSymbol(l2);

	Circle c3(13.4, 52.52, 1.);
		c3.setStrokeColor("red");
		c3.setFillColor("cyan");
	sc.addSymbol(c3);

	Text l3;
		float loc3[2] = {14.4, 52.52};
		l3.setAnchorLocation(loc3);
		l3.setFontSize(10.0f);
		l3.setStrokeWidth(1.0f);
		l3.setText("Berlin");
		l3.setStrokeColor("purple");
		l3.setOpacity(1.0f);
    sc.addSymbol(l3);

	Circle c4(77.208, 28.614, 1.);
		c4.setStrokeColor("red");
		c4.setFillColor("cyan");
	sc.addSymbol(c4);

	Text l4;
		float loc4[2] = {78.208, 28.614};
		l4.setAnchorLocation(loc4);
		l4.setFontSize(10.0f);
		l4.setStrokeWidth(1.0f);
		l4.setText("New Delhi");
		l4.setStrokeColor("purple");
		l4.setOpacity(1.0f);
    sc.addSymbol(l4);
	
	// add the world map
	vector<Country>  wm_data = ds.getWorldMapData();
	WorldMap wm (wm_data);
	bridges.setMap(wm);

	// visualize
	bridges.setDataStructure(sc);
	bridges.visualize();
	
	return 0;
}

File: tutorials/testing/java/tut_worldmap_p4.java
import java.util.ArrayList;
import java.util.Map;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.Country;
import bridges.base.WorldMap;
import bridges.base.Color;
import bridges.base.Rectangle;
import bridges.base.Circle;
import bridges.base.Text;
import bridges.base.SymbolCollection;


// This tutorial illustrates how to access the data of the World map with 
// boundaries; the countries are filled with default color
public class tut_worldmap_p4 {
	public static void main(String[] args) throws Exception {

		// create Bridges object
		Bridges bridges = new Bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", 
												"YOUR_API_KEY");
		// set title, description
		bridges.setTitle("Tutorial: Demonstrating overlaying World Map with Cities at their locations");
		bridges.setDescription("Uses symbols to represent city locations");

		// get data source object
		DataSource ds = bridges.getDataSource();


		// use symbols to represent cities
		SymbolCollection sc = new SymbolCollection();

		// circle properties
		Color red = new Color ("red");
		Color blue = new Color ("blue");

		// label properties
		float fsize = 12.f, str = 1.0f; 
		Color scol = new Color("purple");
		


		// charlotte
		Circle  c1 = new Circle(-80.8431f, 35.2271f, 1.f);
			c1.setStrokeColor(red);
			c1.setFillColor (blue);
		sc.addSymbol(c1);

		Text l1 = new Text();
			l1.setAnchorLocation(-79.8431f, 35.2271f);
			l1.setFontSize(fsize);
			l1.setStrokeColor(scol);
			l1.setText("Charlotte");
		sc.addSymbol(l1);

		// paris
		Circle  c2 = new Circle(2.3514f, 48.8576f, 1.f);
			c2.setStrokeColor(red);
			c2.setFillColor (blue);
		sc.addSymbol(c2);
		Text l2 = new Text();
			l2.setAnchorLocation(3.3514f, 48.8576f);
			l2.setFontSize(fsize);
			l2.setStrokeColor(scol);
			l2.setText("Paris");
		sc.addSymbol(l2);

		// berlin
		Circle  c3 = new Circle(13.4f, 52.52f, 1.f);
			c3.setStrokeColor(red);
			c3.setFillColor (blue);
		sc.addSymbol(c3);
		Text l3 = new Text();
			l3.setAnchorLocation(14.4f, 52.52f);
			l3.setFontSize(fsize);
			l3.setStrokeColor(scol);
			l3.setText("Berlin");
		sc.addSymbol(l3);
		// delhi
		Circle  c4 = new Circle(77.208f, 28.614f, 1.f);
			c4.setStrokeColor(red);
			c4.setFillColor (blue);
		sc.addSymbol(c4);
		Text l4 = new Text();
			l4.setAnchorLocation(78.208f, 28.614f);
			l4.setFontSize(fsize);
			l4.setStrokeColor(scol);
			l4.setText("New Delhi");
		sc.addSymbol(l4);

		// get us map data - states and boundaries
		ArrayList<Country> wm_data = ds.getWorldMapData ();

		WorldMap wm = new WorldMap(wm_data);
		bridges.setMap(wm);
		bridges.setLabelFlag(true);

		// set data structure
		bridges.setDataStructure(sc);

		// visualize
		bridges.visualize();
	}
}
